tests/scripts/build-release.sh

```diff
@@ -12,18 +12,24 @@ function  build() {
     build/run make mod.check
 }
 
-if [[ ${BRANCH_NAME} =~ release- ]]; then
-    build
-    build/run make -C build/release build BRANCH_NAME=${BRANCH_NAME} TAG_WITH_SUFFIX=true GIT_API_TOKEN=${GIT_API_TOKEN}
-    git status &
-    git diff &
-    build/run make -C build/release publish BRANCH_NAME=${BRANCH_NAME} TAG_WITH_SUFFIX=true AWS_ACCESS_KEY_ID=${AWS_USR} AWS_SECRET_ACCESS_KEY=${AWS_PSW} GIT_API_TOKEN=${GIT_API_TOKEN}
-elif [[ ${BRANCH_NAME} =~ master|tag ]]; then
+function publish_and_promote() {
     build
     build/run make -C build/release build BRANCH_NAME=${BRANCH_NAME}  GIT_API_TOKEN=${GIT_API_TOKEN}
     git status &
     git diff &
     build/run make -C build/release publish BRANCH_NAME=${BRANCH_NAME} AWS_ACCESS_KEY_ID=${AWS_USR} AWS_SECRET_ACCESS_KEY=${AWS_PSW} GIT_API_TOKEN=${GIT_API_TOKEN}
     # automatically promote the master builds
-    build/run make -C build/release promote BRANCH_NAME=master CHANNEL=master AWS_ACCESS_KEY_ID=${AWS_USR} AWS_SECRET_ACCESS_KEY=${AWS_PSW}           
-fi
+    if [[ ${GITHUB_REF} =~ master ]]; then
```

Comment:
I have do `=~` since ${GITHUB_REF} is `refs/heads/<branch_name>, and for tags it is refs/tags/<tag_name>`
for more details doc [link](https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#github-context)

Reply:
How about if the condition sets a variable, then you can run a common `build/run` command instead of duplicating the whole command?

Reply:
can you explain this bit more?

Reply:
Something like this:
```
if master then
  CHANNEL=master
else
  CHANNEL=release
fi
build/run make -C build/release promote BRANCH_NAME=${BRANCH_NAME} CHANNEL=${CHANNEL} AWS_ACCESS_KEY_ID=${AWS_USR} AWS_SECRET_ACCESS_KEY=${AWS_PSW}
```

Reply:
okay...will do the changes. Thanks

Reply:
done.